docs/grammar/SolidityParser.g4

```diff
@@ -156,10 +156,11 @@ overrideSpecifier: Override (LParen overrides+=identifierPath (Comma overrides+=
  * Depending on the context in which the function is defined, further restrictions may apply,
  * e.g. functions in interfaces have to be unimplemented, i.e. may not contain a body block.
  */
-contractFunctionDefinition
+functionDefinition[boolean free]
 locals[
 	boolean visibilitySet = false,
 	boolean mutabilitySet = false,
+	boolean suffixSet = false,
```

Comment:
I'm just wondering now, why we don't use the `free` parameter.. so this seem to be creating the same `locals` for free function definitions and not-free functions

Reply:
Because having these locals does not hurt even if they're not used. And, honestly, I'm not even sure if they can be declared conditionally.